pkg/dockerregistry/server/manifestservice.go

```diff
@@ -219,6 +230,55 @@ var manifestInflight = make(map[digest.Digest]struct{})
 // manifestInflightSync protects manifestInflight
 var manifestInflightSync sync.Mutex
 
+// fillImageMetadata fills metadata for image if needed. The metadata is filled by the master API if not
+// already filled and if the maniest and config blobs are sent together with the image. The registry needs to
+// fill the metadata only for schema 1 manifests that don't contain image sizes. Only the registry is capable
+// of fetching and correcting blob sizes.
```

Comment:
In general this is false, anybody with valid credentials can use HEAD to fetch blob sizes.

Reply:
Do you want to rephrase just the comment or do you suggest to make the master fetch the sizes from the registry?

Reply:
Rewritten.